src/vivarium/framework/results/interface.py

```diff
@@ -140,26 +161,93 @@ def register_binned_stratification(
     # Observation-related methods #
     ###############################
 
-    # TODO: This should implement the most basic Observation registration
-    # def register_observation(self, ...):
-    #     pass
-
-    def register_adding_observation(
+    def register_stratified_observation(
```

Comment:
We have discussed at length whether the interface should have a different registration function per observation type or a single register_observation function that takes a "type" arg. That's a bit tricky to implement but might be cleaner? idk - I'm open to thoughts